independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -95,6 +97,12 @@
     private final boolean removeFinalForProxyableMethods;
     private final boolean jtaCapabilities;
 
+    private Function<AnnotationInstance, Collection<AnnotationInstance>> singletonFunction = annotationInstance -> Collections
```

Comment:
These could be final... but it's just a minor.

Reply:
These will be removed altogether accordingly to your earlier suggestion.